invenio_app_ils/records/related/__init__.py

```diff
@@ -0,0 +1,199 @@
+        """Initialize related records proxy.
+
+        :param parent: Parent record
+        """
+        self.parent = parent
+        self.changed_related_records = []
+
+    @property
+    def editions(self):
+        """Get related editions."""
+        return self.get_related(RelatedRecords.edition_relation())
+
+    @property
+    def languages(self):
+        """Get related languages."""
+        return self.get_related(RelatedRecords.language_relation())
+
+    def add_edition(self, record):
+        """Add a new related edition."""
+        self.add(record, RelatedRecords.edition_relation())
+
+    def add_language(self, record):
+        """Add a new related language."""
+        self.add(record, RelatedRecords.language_relation())
+
+    def add(self, record, relation):
+        """Add a new related record."""
+        self._add_record(
+            record,
+            relation,
+            parent_node=self._parent_node(relation)
+        )
+
+    def _add_record(self, record, relation, parent_node=None):
+        """Add related record."""
+        if record == self.parent:
+            raise RelatedRecordError("Cannot add itself as a related record.")
+        record_node = get_node(record, relation)
+        if record_node.is_parent or record_node.is_child:
+            raise RelatedRecordError(
+                "Failed to add related record because it already has relations"
+            )
+        if parent_node is None:
+            parent_node = get_node(self.parent, relation)
+        self.dump_add(self.parent, record, relation, update_related=True)
+        parent_node.insert_child(record.pid)
+
+    def remove_edition(self, record):
+        """Remove related edition."""
+        self.remove(record, RelatedRecords.edition_relation())
+
+    def remove_language(self, record):
+        """Remove related language."""
+        self.remove(record, RelatedRecords.language_relation())
+
+    def remove(self, record, relation):
+        """Remove existing related record."""
+        self._remove_record(
+            record,
+            relation,
+            parent_node=self._parent_node(relation)
+        )
+
+    def _remove_record(self, child, relation, parent_node=None):
+        """Remove related record with a relation."""
+        if parent_node is None:
+            parent_node = get_node(self.parent, relation)
+        parent_pid = parent_node.pid
+        parent = self.parent
+        is_parent_equal = parent_pid.pid_value == child[child.pid_field] and \
+            parent_pid.pid_type == child._pid_type
+        if is_parent_equal:
+            # Trying to remove parent
+            # TODO: remove old relations and copy over relations to self.parent
+            records_cfg = current_app.config["RECORDS_REST_ENDPOINTS"]
+            record_name = records_cfg[child._pid_type]["record_class"].__name__
+            raise RelatedRecordError(
+                ("Failed to remove {} relation for {} {} because it is the "
```

Comment:
I think this message should be maybe more a log.
The user does not really now about parent/child and IMHO is going to be confused.
What about something like (please, fix my english):
`Cannot remove the language relation. Please, remove this relation by editing Document 1`